### PR TITLE
Enable Jenkins pipeline

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Router: Create a new Micro service](https://tools.hmcts.net/jira/browse/BPS-794)

### Change description ###

Includes some generic fixes and reviews before applying the `Jenkinsfile_CNP` which enables pipeline builds.

The desired outcome is to get green build at least on PR so feature of protecting PR merges can be enabled

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  after('smoketest:aat') {
+    sh "rm -rf infrastructure/"
+  }
+
+  enableDockerBuild()
+  enableAksStagingDeployment()
+  installCharts()
+  enableSlackNotifications(channel)
+  disableLegacyDeployment()
```
Comment (nitinprabhu) [inline Jenkinsfile_CNP]: do we need this ? It was never deployed on ASE ? but never know for DNS we may need it

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+def channel = '#bsp-build-notices'
+GradleBuilder builder = new GradleBuilder(this, product)
+
+withPipeline(type, product, component) {
+  after('test') {
+    builder.gradle('integration')
+  }
+
+  after('smoketest:aat') {
+    sh "rm -rf infrastructure/"
```
[the remark this answers is not in the document]
Comment (doncem) [reply Jenkinsfile_CNP]: true. left/included this for for super safety i guess :D

File: docker-compose.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ services:
         - http_proxy
         - https_proxy
         - no_proxy
-    image: hmcts/blob-router-service
+    image: reform-scan/blob-router
```
[the remark this answers is not in the document]
Comment (doncem) [reply docker-compose.yml]: perhaps needs full url, right?

File: Dockerfile [hunk 4]
```diff
@@ -4,7 +4,7 @@ ARG APP_INSIGHTS_AGENT_VERSION=2.5.0
 
 FROM busybox as downloader
 
-RUN wget -P /tmp https://github.com/microsoft/ApplicationInsights-Java/releases/download/${APP_INSIGHTS_AGENT_VERSION}/applicationinsights-agent-${APP_INSIGHTS_AGENT_VERSION}.jar
+RUN wget -P /tmp https://github.com/microsoft/ApplicationInsights-Java/releases/download/2.5.0/applicationinsights-agent-2.5.0.jar
```
Comment (rkondratowicz) [inline Dockerfile]: please explain

Comment (doncem) [reply Dockerfile]: not sure which part to explain so will try to explain as fully as i can.

commit message says "Fix `wget` command in Dockerfile". diff suggests that previous version did not work. ARG setting is not used in this command line run hence need hardcoded yet again.

it is way easier to change 3 places in this file than downloading jar yourself and sticking in repository.

inspected such error on different application - sample app just moments ago

Comment (nitinprabhu) [reply Dockerfile]: Do we need APP_INSIGHTS_AGENT_VERSION any more than ?

Comment (doncem) [reply Dockerfile]: Can't see in jenkins library. If suppose to be used only by that library then it doesn;t serve any value right now - single point of use below in same file. I can test it out once pipeline is ready in Jenkins. For now - let's keep it as all other projects (including template) has it

**Pull request merged**